### PR TITLE
WIP: feat: md chatbar

### PR DESCRIPTION
**Pull request by molimauro**
**What this PR does** 📖
Handle the MD in the chatbar

**Which issue(s) this PR fixes** 🔨
* Cursor position
* Breaklines (shift + Enter)
* MD in the chatbar

**Special notes for reviewers** 🗒️
Still a WIP do not merge please

**Additional comments** 🎤
This PR is to have feedbacks on this feature. Being a major feature, there are a lot of things in the TODO list:

* Paste currently not working;
* Code blocks with \`\`\`code\`\`\` and `highlight.js` still not present;
* Input sanitization;
* Still have to be tweaked with all the features in the current chatbar (commands, emojis, etc..);
* CSS improvement.

I'd like to know which of these points should be tackled first in order to organize my work. Thanks! :)

### CODE REVIEW

**Review by iltumio — COMMENTED** [review]
it works very well from my point of view. The feeling is great. We need to remove all the html code from the characters count and try to enable commands and emoji, fix some styles, then I think we can release it

Schermata 2021-12-18 alle 01 04 57

File: layouts/chat.vue [hunk 1]
```diff
@@ -107,7 +107,7 @@ export default Vue.extend({
             (friend) => friend.address === this.$route.params.address,
           )
       return recipient
-    }
+    },
```
[the remark this answers is not in the document]
Comment (stavares843) [reply layouts/chat.vue:110]: yes this was we choose, adding a '','' on all of them

https://discord.com/channels/773445116572073995/859670812856418314/914554786202935296

[the remark this answers is not in the document]
Comment (stavares843) [reply layouts/chat.vue:110]: this seems the only modification on the whole file, and the linter only runs on the staged files, so probably this wasn't the linter

**Comment by RetroPronghorn** [comment]
This PR is great!

**Review by WanderingHogan — COMMENTED** [review]
I like the direction you're taking and look forward to seeing it when it's done

**Comment by RetroPronghorn** [comment]
@molimauro just checking in on this, what is the status for this ticket?

**Comment by molimauro** [comment]
I removed _marked_ for _simple-markdown_, this will allow us to customize more commands in the future as well as @everyone, @chat commands etc.. Tomorrow I'll upload the PR so that you can check the status ;) Now I'm working on the highlightjs part that is a bit tricky

**Review by RetroPronghorn — APPROVED** [review]

**Comment by WanderingHogan** [comment]
@molimauro can you close this draft PR to get it out of the queue until you are ready? Can you also give an update, @RetroPronghorn is pretty excited about this feature and is excited it might be going in soon.

**Comment by molimauro** [comment]
Of course! I'm integrating all the existing functionalities (drop, upload, send, etc.) into the new chatbar and styling it a little bit. I've paused the highlightjs part in order to merge this asap

**Pull request closed without merging**